.github/workflows/pytest.yml

```diff
@@ -14,6 +14,27 @@ env:
   DAGSTER_HOME: /home/runner/pudl-work/dagster_home/
 
 jobs:
+  change_filter:
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: read
+    outputs:
+      run_code_checks: ${{ github.event_name =='workflow_dispatch' || (steps.filter.outputs.code == 'true' && github.event.merge_group) }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - uses: dorny/paths-filter@v3
+        id: filter
+        with:
+          # this will raise a warning, but it works:  https://github.com/dorny/paths-filter/issues/225
+          predicate-quantifier: "every"
+          filters: |
+            code:
+              - '!docs/**.rst'
+              - '!docs/**.md'
+              - '!README.rst'
```

Comment:
I wanted to be fairly conservative here so we don't accidentally skip the code checks when code has changed - this says that "if any file that has changed is not the README, or a md/rst within the docs folder, that counts as a code change"

Reply:
There are fewer of these files and they get updated less frequently but we might also want to include:

docs/**.pdf
docs/**.html

What about docs/**.rst.jinja? If there haven't been changes to any python files, but the Jinja template content has changed, is there a risk of something that needs to get tested not getting tested? Since those templates will get exercised in the docs build.
Maybe need to make sure that this gets another look when the new metadata stuff is merged in, in case there are other things that need to be included / ignored.

Reply:
Yeah, I think `**.rst.jinja` should be OK also:

- if the change is only to `**.rst.jinja` it should get caught in docs build
- if the change is to something else that feeds into a custom macro or something, it will trigger the code checks.